src/github/github.installation.controller.ts

```diff
@@ -0,0 +1,203 @@
+    }
+  }
+
+  @Put(':id')
+  async update(
+    @Param('id') id: string,
+    @Body() updateOrgDto: CreateOrgDto,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.update(id, updateOrgDto);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Delete(':id')
+  async remove(@Param('id') id: string): Promise<void> {
+    try {
+      return this.githubInstallationService.remove(id);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('is-allowed/:organizationName')
+  async isOrganizationAllowed(
+    @Param('organizationName') organizationName: string,
+  ): Promise<boolean> {
+    try {
+      return this.githubInstallationService.isOrganizationAllowed(
+        organizationName,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('is-user-allowed')
+  async isUserAllowed(
+    @Query('organizationName') organizationName: string,
+    @Query('githubUsername') githubUsername: string,
+  ): Promise<boolean> {
+    try {
+      return this.githubInstallationService.isUserAllowed(
+        organizationName,
+        githubUsername,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put('add-user/:organizationName')
+  async addAllowedUser(
+    @Param('organizationName') organizationName: string,
+    @Query('githubUsername') githubUsername: string,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.addAllowedUser(
+        organizationName,
+        githubUsername,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put('remove-user/:organizationName')
+  async removeAllowedUser(
+    @Param('organizationName') organizationName: string,
+    @Query('githubUsername') githubUsername: string,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.removeAllowedUser(
+        organizationName,
+        githubUsername,
+      );
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
```

Comment:
PullProofAI

Consider consolidating error handling logic to avoid repetition.
```diff
-       if (error.status) {
-         throw new HttpException(error.message, error.status);
-       } else {
-         throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
-       }
+       this.handleError(error);
```



<!-- This is an auto-generated comment by PullProofAI -->